src/applications/disability-benefits/all-claims/utils/submit.js

```diff
@@ -752,6 +757,309 @@ export const addFileAttachments = formData => {
+ * @param {string} [obj.description] - Description property if obj is an object
+ * @returns {boolean} True if a valid non-empty description exists
+ */
+const hasValidDescription = obj => {
+  if (!obj) return false;
+
+  const description = typeof obj === 'string' ? obj : obj?.description;
+  return (
+    description &&
+    typeof description === 'string' &&
+    description.trim().length > 0
+  );
+};
+
+/**
+ * Filters conditions to only include those explicitly checked (true).
+ * Removes conditions that are false or unchecked from the submission.
+ *
+ * @param {Object} conditions - Object with condition names as keys and boolean values
+ * @returns {Object} Object containing only conditions set to true
+ */
+const filterCheckedConditions = conditions => {
+  if (!isPlainObject(conditions)) return {};
+
+  return Object.keys(conditions).reduce((filtered, condition) => {
+    if (conditions[condition] === true) {
+      return { ...filtered, [condition]: true };
+    }
+    return filtered;
+  }, {});
+};
+
+/**
+ * Determines if toxic exposure object contains no meaningful data.
+ * Used to decide whether to remove the entire toxic exposure section.
+ *
+ * @param {Object} toxicExposure - The toxic exposure object to check
+ * @returns {boolean} True if object is empty or contains only empty/null values
+ */
+const isEmptyToxicExposure = toxicExposure => {
+  return Object.keys(toxicExposure).every(key => {
+    const value = toxicExposure[key];
+    return (
+      !value || (typeof value === 'object' && Object.keys(value).length === 0)
+    );
+  });
+};
+
+/**
+ * Creates a toxic exposure object with only 'none' condition selected.
+ * Used when veteran indicates they have no toxic exposure conditions.
+ *
+ * @returns {Object} Object with conditions containing only none: true
+ */
+const createNoneOnlyCondition = () => ({
+  conditions: { none: true },
+});
+
+/**
+ * Cleans exposure details by removing unselected items and orphaned fields.
+ * Handles all related fields for an exposure type based on the mapping configuration.
+ *
+ * @param {Object} toxicExposure - The toxic exposure object containing selections and details
+ * @param {string} exposureType - Key for the exposure selections (e.g., 'gulfWar1990')
+ * @param {Object} mapping - The mapping configuration for this exposure type
+ * @returns {Object} New toxic exposure object with cleaned details and related fields
+ */
+const cleanExposureDetails = (toxicExposure, exposureType, mapping) => {
+  const { detailsKey, otherLocationsKey } = mapping;
+  const result = { ...toxicExposure };
+
+  // If details exist but main selection is missing, remove the details
+  if (detailsKey && result[detailsKey] && !result[exposureType]) {
+    delete result[detailsKey];
+    return result;
+  }
+
+  // If main selection exists but has no selected values, remove it and related fields
+  if (result[exposureType] && !hasSelectedConditions(result[exposureType])) {
```

Comment:
I'm confused here: According to L836, shouldn't the param be an Object with condition selections as boolean values?